src/main/scala/com/mozilla/telemetry/pings/package.scala

```diff
@@ -0,0 +1,225 @@
+                   normalizedChannel: String,
+                   os: String,
+                   sampleId: Option[Double],
+                   sourceName: Option[String],
+                   sourceVersion: Option[Int],
+                   submissionDate: String,
+                   telemetryEnabled: Option[Boolean],
+                   // Common fields preparsed by hindsight
+                   `environment.build`: Option[EnvironmentBuild],
+                   `environment.settings`: Option[Settings],
+                   `environment.system`: Option[System],
+                   `environment.profile`: Option[Profile],
+                   `environment.addons`: Option[Addons],
+                   // Main ping fields preparsed by hindsight
+                   `payload.simpleMeasurements`: JValue,
+                   `payload.keyedHistograms`: JValue,
+                   `payload.histograms`: JValue,
+                   `payload.info`: JValue
+                 )
+
+  case class CrashPayload(
+                           crashDate: String,
+                           processType: Option[String],
+                           hasCrashEnvironment: Option[Boolean],
+                           metadata: Option[Map[String, String]],
+                           version: Option[Int]
+                         )
+
+  case class CrashPing(
+                        application: Application,
+                        clientId: Option[String],
+                        payload: CrashPayload,
+                        // Environment is omitted it's partially available under meta
+                        meta: Meta
+                      ){
+
+    def isMainCrash(): Boolean = {
+      payload.processType.getOrElse("main") == "main"
+    }
+
+    def timestamp(): Timestamp = {
+      new Timestamp(this.meta.Timestamp / 1000000)
+    }
+  }
+
+  case class PayloadInfo(subsessionLength: Option[Int])
+
+  case class MainPing(
+                       application: Application,
+                       clientId: Option[String],
+                       // Environment omitted because it's mostly available under meta
+                       meta: Meta
+                     ){
+    def getCountHistogramValue(histogram_name: String): Int = {
+      try {
+        this.meta.`payload.histograms` \ histogram_name \ "values" \ "0" match {
+          case JInt(count) => count.toInt
+          case _ => 0
+        }
+      } catch { case _: Throwable => 0 }
+    }
+
+    def getCountKeyedHistogramValue(histogram_name: String, key: String): Int = {
+      try {
+        this.meta.`payload.keyedHistograms` \ histogram_name \ key \ "values" \ "0" match {
+          case JInt(count) => count.toInt
+          case _ => 0
+        }
+      } catch { case _: Throwable => 0 }
+    }
+
+
+    def usageHours(): Float = {
+      try {
+        this.meta.`payload.info` \ "subsessionLength" match {
+          case JInt(length) => Math.min(25, Math.max(0, length.toFloat / 3600))
```

Comment:
I didn't touch this formula here. This comes from [crash_aggregates](https://github.com/mozilla/telemetry-batch-view/blob/master/src/main/scala/com/mozilla/telemetry/views/CrashAggregateView.scala#L326), which inherited it from [crash_aggregator](https://github.com/mozilla/moz-crash-rate-aggregates/blob/master/crash_rate_aggregates/crash_aggregator.py#L163). It seems to be an arbitrary cap to normalize the maximum usage hours that a ping can represent. I'm not sure what kind of analysis led to 25 vs 24.